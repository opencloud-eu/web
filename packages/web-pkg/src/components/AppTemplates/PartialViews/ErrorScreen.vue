<template>
  <div class="oc-text-center oc-flex oc-flex-center oc-flex-middle oc-height-1-1">
    <oc-icon size="xxlarge" name="error-warning" fill-type="line" />
    <p v-if="message" class="text-xl">{{ message }}</p>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
export default defineComponent({
  name: 'ErrorScreen',
  props: {
    /**
     * Error message passed from external app
     */
    message: {
      default: '',
      type: String,
      required: false
    }
  }
})
</script>
