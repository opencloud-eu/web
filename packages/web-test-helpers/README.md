# web-test-helpers

This packages provides utilities for unit testing within the OpenCloud app ecosystem.

## Installation

Depending on your package manager, run one of the following commands:

```
$ npm install @opencloud-eu/web-test-helpers --save-dev

$ pnpm add -D @opencloud-eu/web-test-helpers

$ yarn add @opencloud-eu/web-test-helpers --dev
```

Make sure that you have `vue`, `@vue/test-utils`, and `@opencloud-eu/web-pkg` as dev dependencies of your app since those are peer dependencies of this package.
