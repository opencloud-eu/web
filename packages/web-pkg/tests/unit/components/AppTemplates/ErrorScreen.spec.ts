import ErrorScreen from '../../../../src/components/AppTemplates/PartialViews/ErrorScreen.vue'
import { defaultPlugins, mount } from '@opencloud-eu/web-test-helpers'

describe('The external app error screen component', () => {
  test('displays an icon and a paragraph', () => {
    const wrapper = mount(ErrorScreen, {
      props: {
        message: 'Error when loading the application'
      },
      global: {
        stubs: {
          OcIcon: true
        },
        plugins: [...defaultPlugins()]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
