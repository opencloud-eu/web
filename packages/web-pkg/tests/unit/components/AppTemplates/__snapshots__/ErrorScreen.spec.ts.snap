// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The external app error screen component > displays an icon and a paragraph 1`] = `
"<div class="oc-text-center oc-flex oc-flex-center oc-flex-middle oc-height-1-1">
  <oc-icon-stub accessiblelabel="" color="" filltype="line" name="error-warning" size="xxlarge" type="span"></oc-icon-stub>
  <p class="oc-text-xlarge">Error when loading the application</p>
</div>"
`;
